src/entity/Entity.php

```diff
@@ -666,6 +666,7 @@ protected function entityBaseTick(int $tickDiff = 1) : bool{
 				foreach ($block->getCollisionBoxes() as $blockBox) {
 					if ($entityBox->intersectsWith($blockBox)) {
 						$block->onEntityCollide($this, Facing::opposite($face));
+						$hasUpdate = true;
```

Comment:
This shouldn't be unconditionally set. You should add a return value to `onEntityCollide` so it knows whether to set this or not.

Reply:
I also just realized this is in entityBaseTick...